### DISCUSSION TITLE
SPF validation status in HTTP route body

### DISCUSSION THREAD

**Discussion by ghost** [postalserver/postal#1744] [Feature suggestions]
Would it be possible to add a field in the [payload to HTTP routes](https://docs.postalserver.io/developer/http-payloads#the-processed-payload) for the actual SPF validation status of an email instead of just having `spam_status`? For my use case, I need to determine whether I can trust the "From" address and display it to the user, or if I should show "Sender" instead (if the SPF validation fails and "From" can't be trusted).

Right now, there seems to be no way for me to reliably determine this. I've considered trying to check SPF myself, but the only option I can find is [node-spf-check](https://github.com/mediamonks/node-spf-check/), which often returns an unsatisfactory result and requires the IP address that the email was sent from, and I can't find any way to get that from Postal either.

Determining whether the "From" address was spoofed beyond a simple "this message may be spam" warning seems pretty important for most applications involving email. Is there something I'm missing? If there's no way for me to determine this, I really think it should be added to the data sent to HTTP routes.

**Comment by willpower232** [comment]
If you have spam filtering enabled then the `spam_status` field should be correct as I think both rspamd and spamassassin would do that verification. I'm not aware of Postal doing any direct SPF check itself so it wouldn't know what to tell you.

**Comment by ghost** [comment]
The problem is that I want to reject spam, but I want to process messages that simply fail SPF validation since in my experience a lot of automated emails don't do SPF correctly, so I'd just display them with the "Sender" address.

It seems like an odd design to have "this message contains a link to a suspicious website" and "this message is falsely claiming to be sent from a bank's email address" shown as the same ambiguous spam status, especially if that's the only way to detect whether the message is actually sent from that address to the receiving server, but I understand that it's not directly Postal's fault.

**Reply by willpower232** [reply]
Postal is designed for transactional purposes so if you're expecting to handle incoming email like a regular email inbox then it won't quite meet your needs without some extra work on whatever you're using to send emails with Postal.
